crates/bevy_sprite/src/render/mod.rs

```diff
@@ -186,6 +187,20 @@ pub struct ExtractedSprite {
     pub flip_y: bool,
 }
 
+#[derive(Clone, Copy)]
+pub struct Transform2d {
+    pub translation: Vec2,
+    pub scale: Vec2,
+    pub rotation: Quat,
```

Comment:
it feels wrong to keep a `Quat` for rotations, but I'm not sure if keeping only two axis of rotations would be enough here. I feel it would, which would probably make it faster?

Reply:
Don't we only need a single 2D rotation axis?

Reply:
heh 3d rotations in 2d are one of the things where I give up usually...

Reply:
I think we can use [`Quat::mul_vec3`](https://docs.rs/glam/latest/glam/f32/struct.Quat.html#method.mul_vec3) with `Vec3::X`, and truncate that. We then get the angle from that method, e.g. [`Vec2::angle_between`](https://docs.rs/glam/latest/glam/f32/struct.Vec2.html#method.angle_between) with `Vec2::X`.

Reply:
> I think we can use [`Quat::mul_vec3`](https://docs.rs/glam/latest/glam/f32/struct.Quat.html#method.mul_vec3) with `Vec3::X`

that's exactly what `local_x()` does, so it's simply
```
Vec2::X.angle_between(transform.local_x().truncate())
```
"Get the angle between the global x-axis and the local x unit-vector"